File: 0154-find-minimum-in-rotated-sorted-array-ii/0154-find-minimum-in-rotated-sorted-array-ii.py
class Solution(object):
    def findMin(self, nums):
       min = nums[0]
       for i in range(1,len(nums)):
            j=-i
            if nums[i]<min:
                min=nums[i]
            elif nums[j]<min:
                min=nums[j]
            
            if i>len(nums)/2:
                break
        
       return min

        

File: 0154-find-minimum-in-rotated-sorted-array-ii/0154-find-minimum-in-rotated-sorted-array-ii.java
class Solution {
    public int findMin(int[] nums) {
        int left=0;
        int right=nums.length-1;
        int mid;
        if(nums.length==1)
            return nums[0];

        while(left<right){
            mid=(left+right)/2;
            if(nums[mid]==nums[right])
                right-=1;
            else if(nums[mid]>nums[right])
                left=mid+1;
            else
                right=mid;
        }
        return nums[left];
    }
}